molecule/test/functional/conftest.py

```diff
@@ -277,6 +282,30 @@ def supports_docker():
     return True
 
 
+@pytest.helpers.register
+@util.lru_cache()
+def supports_podman():
+    # Returns true if podman is supported and working
+    # Returns false if podman in not supported
+    # Calls pytest.fail if podman appears to be broken
+    if min_ansible("2.8.6") and platform.system() != 'Darwin':
+        podman = get_podman_executable()
+        if podman:
+            result = util.run([podman, "info"], stdout=PIPE, universal_newlines=True)
+            if result.returncode != 0:
+                LOG.error(
+                    "Error %s returned from `podman info`: %s",
+                    result.returncode,
+                    result.stdout,
+                )
+                pytest.fail(
+                    "Cannot run podman tests with a broken podman installation."
+                )
+                return False
+            return True
```

Comment:
because of nested "if"s it's not so clear what this "true" is related to, and which conditions are related to multiple "return false"